posts/Fabio/2018-03-28-hibernate-ogm-5-3-1-Final-released.adoc

```diff
@@ -0,0 +1,118 @@
+= "Hibernate OGM 5.3.1.Final is out"
+Fabio Massimo Ercoli
+:awestruct-tags: [ "Hibernate OGM", "Releases" ]
+:awestruct-layout: blog-post
+---
+
+We're happy to announce the release of:
+http://hibernate.org/ogm/releases/5.3/#get-it[Hibernate OGM 5.3.1.Final]
+
+== What's new compared to 5.3.0.Final?
+
+This ia a fix release for native queries pagination and projection issues.
+Now pagination works perfectly for MongoDB native queries that use aggregates.
+You can see a complete list of the changes on the
+https://hibernate.atlassian.net/secure/ReleaseNote.jspa?projectId=10160&version=31644[release notes].
+
+This version is fully compatible with 5.3.0.
+
+== Components upgrade
+
+ * Hibernate ORM 5.2.16.Final
+
+== Projection and addEntity are not allowed in the same query
+
+It is not anymore possible to execute a native query with projection
+and extract the returned values as an entity list.
+Given the entity:
+
+====
+[source,java]
+----
+@Entity
+public class Movie {
+
+	@Id
+	private Integer id;
+	private String name;
+	private String author;
+	private Integer year;
+
+	public Movie() {
+	}
+}
+----
+====
+
+If we try to get all movies using **projection** and **addEntity**:
+
+====
+[source,java]
+----
+session.createNativeQuery( "db.Movie.find( {}, { 'id' : 1, 'name' : 1  } )" )
+	.addEntity( Movie.class )
+	.uniqueResult();
+----
+====
+
+An exception will be thrown and this message will be presented on log:
+
+====
+OGM000091: Projection and addEntity are not allowed in the same query on <Movie>
+====
+
+It is legal instead using **projection** without **addEntity**:
+
+====
+[source,java]
+----
+session.createNativeQuery( "db.Movie.find( {}, { 'id' : 1, 'name' : 1  } )" )
+	.uniqueResult();
+----
+====
+
+Or using **addEntity** without **projection** :
+
+====
+[source,java]
+----
+session.createNativeQuery( "db.Movie.find( {} )" )
+	.addEntity( Movie.class )
+	.uniqueResult();
+----
+====
+
+This is valid for all dialects that support native queries,
+at the moment of writing those are MongoDB and Neo4j dialects.
```

Comment:
This is true for all dialects supporting native queries, which at the moment of writing those lines are the MongoDB and Neo4j dialects.